pkg/controllers/clusterresourcebindingwatcher/watcher_integration_test.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+package clusterresourcebindingwatcher
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	testCRPName                      = "test-crp"
+	testCRBName                      = "test-crb"
+	testResourceSnapshotName         = "test-rs"
+	testSchedulingPolicySnapshotName = "test-sps"
+	testTargetCluster                = "test-cluster"
+	testReason                       = "testReason"
+
+	eventuallyTimeout    = time.Second * 10
+	consistentlyDuration = time.Second * 10
+	interval             = time.Millisecond * 250
+)
+
+// This container cannot be run in parallel with other ITs because it uses a shared fakePlacementController.
+var _ = Describe("Test ClusterResourceBinding Watcher - create, delete events", Serial, func() {
+	var crb *fleetv1beta1.ClusterResourceBinding
+	It("When creating, deleting clusterResourceBinding", func() {
+		By("Creating a new clusterResourceBinding")
+		crb = clusterResourceBindingForTest()
+		Expect(k8sClient.Create(ctx, crb)).Should(Succeed(), "failed to create cluster resource binding")
+
+		By("Checking placement controller queue")
+		consistentlyCheckPlacementControllerQueueIsEmpty("watcher should ignore the create event and not enqueue the request into the placementController queue")
+
+		Expect(k8sClient.Get(ctx, types.NamespacedName{Name: testCRBName}, crb)).Should(Succeed(), "failed to get cluster resource binding")
+
+		By("Deleting clusterResourceBinding")
+		Expect(k8sClient.Delete(ctx, crb)).Should(Succeed(), "failed to delete cluster resource binding")
+
+		By("Checking placement controller queue")
+		consistentlyCheckPlacementControllerQueueIsEmpty("watcher should ignore the delete event and not enqueue the request into the placementController queue")
+	})
+})
+
+// This container cannot be run in parallel with other ITs because it uses a shared fakePlacementController.
+var _ = Describe("Test ClusterResourceBinding Watcher - update events", Serial, func() {
+	var crb *fleetv1beta1.ClusterResourceBinding
+	Context("When updating clusterResourceBinding spec, status", func() {
+		BeforeEach(func() {
+			fakePlacementController.ResetQueue()
+			By("Creating a new clusterResourceBinding")
+			crb = clusterResourceBindingForTest()
+			Expect(k8sClient.Create(ctx, crb)).Should(Succeed(), "failed to create cluster resource binding")
```

Comment:
suggest to resetQueue after the create to make sure the queue is clean before the update.